pages/index.js

```diff
@@ -1,69 +1,95 @@
-import Head from 'next/head'
-import Image from 'next/image'
-import styles from '../styles/Home.module.css'
+import { useState } from "react";
+import { v4 as uuid } from "uuid";
+import TodoItem from "./components/TodoItem";
+import TodoList from "./Components/TodoList";
 
 export default function Home() {
-  return (
-    <div className={styles.container}>
-      <Head>
-        <title>Create Next App</title>
-        <meta name="description" content="Generated by create next app" />
-        <link rel="icon" href="/favicon.ico" />
-      </Head>
+  const [todos, setTodos] = useState([
+    { title: "test Todo", id: uuid(), done: true },
+  ]);
+
+  const [searchQuery, setSearchQuery] = useState("");
 
-      <main className={styles.main}>
-        <h1 className={styles.title}>
-          Welcome to <a href="https://nextjs.org">Next.js!</a>
-        </h1>
+  const handleDeleteTodo = (id) => {
+    setTodos(todos.filter((todo) => todo.id !== id));
+  };
 
-        <p className={styles.description}>
-          Get started by editing{' '}
-          <code className={styles.code}>pages/index.js</code>
-        </p>
+  const handleAddtodo = (event) => {
+    if (event.key === "Enter") {
+      setTodos([
+        ...todos,
+        { id: uuid(), title: event.target.value, done: false },
+      ]);
+      event.target.value = "";
+    }
+  };
 
-        <div className={styles.grid}>
-          <a href="https://nextjs.org/docs" className={styles.card}>
-            <h2>Documentation &rarr;</h2>
-            <p>Find in-depth information about Next.js features and API.</p>
-          </a>
+  // Abhacken von Todos
+  const toggleTodo = (id) => {
+    setTodos(
+      todos.map((todo) => {
+        if (todo.id === id) {
+          return { ...todo, done: !todo.done };
+        }
+        return todo;
+      })
+    );
+  };
 
-          <a href="https://nextjs.org/learn" className={styles.card}>
-            <h2>Learn &rarr;</h2>
-            <p>Learn about Next.js in an interactive course with quizzes!</p>
-          </a>
+  const activeTodos = todos.filter(({ done }) => !done);
+  const doneTodos = todos.filter(({ done }) => done);
+  const filterTodos = todos.filter((todo) =>
+    todo.title.toLowerCase().includes(searchQuery.toLowerCase())
+  );
+
+  // Suche von Todos
+  return (
+    <div className="main">
+      <div>
+        <input
+          onChange={(event) => setSearchQuery(event.target.value)}
```

Comment:
```suggestion
          value={searchQuery}
          onChange={(event) => {
            setSearchQuery(event.target.value);
          }
```

No implicit return.
Missing value (See: https://reactjs.org/docs/forms.html#controlled-components)